pkg/server/vault.go

```diff
@@ -0,0 +1,405 @@
+		}
+	}
+	time.Sleep(1 * time.Second)
+
+	_ = enableKVSecretsEngine(v.client, "secret")
+	//if sErr != nil {
+	//	return &VaultError{Message: fmt.Sprintf("%v", sErr)}
+	//}
+	return nil
+}
+
+// WriteSecret stores a key-value pair in the Vault secrets engine in development mode.
+//
+// This function is intended for use with a local Vault instance. It validates input parameters
+// and writes the specified key-value pair to the given path in Vault.
+//
+// Parameters:
+// - path: The Vault path where the secret should be stored (e.g., "secret/myapp").
+// - key: The name of the key to store in the secret (e.g., "API_KEY").
+// - value: The value associated with the key.
+//
+// Returns:
+// - *VaultError: An error object if the operation fails; otherwise, nil.
+func (v *VaultClient) WriteSecret(path, key, value string) *VaultError {
+	if err := func() *VaultError {
+		switch {
+		case v.client == nil:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorClient)}
+		case len(path) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyPath)}
+		case len(key) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyName)}
+		case len(value) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyData)}
+		default:
+			return nil
+		}
+	}(); err != nil {
+		return err
+	}
+
+	client := v.client
+	// Store key-value
+	secretData := map[string]interface{}{
+		"data": map[string]interface{}{
+			key: value,
+		},
+	}
+	fmt.Println("Saving secret at path:", path, "with key:", key, "and value:", value)
+	_, vErr := client.Logical().Write(path, secretData)
+	if vErr != nil {
+		return &VaultError{Message: fmt.Sprintf(VaultErrorWriting, vErr)}
+	}
+	return nil
+}
+
+// ReadSecret retrieves a specific key's value from the Vault secrets engine.
+//
+// This function reads a stored secret from Vault at the specified path and extracts
+// the requested key's value.
+//
+// Parameters:
+// - path: The Vault path where the secret is stored (e.g., "secret/myapp").
+// - key: The specific key within the secret to retrieve.
+//
+// Returns:
+// - string: The value associated with the key, if found.
+// - *VaultError: An error object if the operation fails or the key does not exist.
+func (v *VaultClient) ReadSecret(path, key string) (string, *VaultError) {
+	if err := func() *VaultError {
+		switch {
+		case v.client == nil:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorClient)}
+		case len(path) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyPath)}
+		case len(key) == 0:
+			return &VaultError{Message: fmt.Sprintf(VaultErrorNoKeyName)}
+		default:
+			return nil
+		}
+	}(); err != nil {
+		return "", err
+	}
+
+	client := v.client
+
+	secret, vErr := client.Logical().Read(path)
+	if vErr != nil || secret == nil {
+		return "", &VaultError{Message: fmt.Sprintf(VaultErrorReadResult, vErr)}
+	}
+
+	// Extract the value
+	data := secret.Data["data"].(map[string]interface{})
+	if valueKey, ok := data[key]; ok {
```

Comment:
Map access needs to be checked. The type cast needs to be checked as well. 
```suggestion
	v, found := secret.Data["data"]
	if !found {
		// do something
	 }
	data, ok := v.(map[string]interface{})
	if !ok {
		// do something about the cast failure
	}
	if valueKey, ok := data[key]; ok {
```

Reply:
fixed on new pull request